src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,85 @@
+package duke;
+
+import java.io.IOException;
+
+import javafx.application.Application;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Scene;
+import javafx.scene.layout.AnchorPane;
+import javafx.stage.Stage;
+
+/**
+ * Ui class containing logic for receiving user input and printing output
+ */
+public class Ui extends Application {
+
+    private static final String GREETINGS = "Hello! I'm Ekud \n" + "What can I do for you?";
+    private final Duke duke = new Duke();
+    private MainWindow mainWindow;
+
+    /**
+     * Starts the GUI.
+     * @param stage The stage parameter set by JavaFX.
+     */
+    @Override
+    public void start(Stage stage) {
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(
+                    Ui.class.getResource("/view/MainWindow.fxml"));
+            AnchorPane ap = fxmlLoader.load();
+            Scene scene = new Scene(ap);
+            stage.setScene(scene);
+            fxmlLoader.<MainWindow>getController().setDuke(duke);
+            mainWindow = fxmlLoader.<MainWindow>getController();
+            mainWindow.setDuke(duke);
+            printGreetings();
+            if (duke.getLoaded()) {
+                print("File successfully loaded!");
+            } else {
+                print("Error parsing load file");
```

Comment:
This can be put under Exception handling instead as it is an error message that should show with an exception. DukeException could also be expanded upon to have multiple more specific exceptions.